### PR TITLE
Swift 5. Updated pods. Moved sign functions to Keychain

### PR DESCRIPTION
**Pull request by ypopovych**
Updated PR with compatibility with Swift 5 only.

Also moved `EthereumPrivateKey` and `EthereumPublicKey` to own package.
For someone who wants to use the library with own signature/key management parts.
Enabled by default for CocoaPods.

### CODE REVIEW

**Comment by ypopovych** [comment]
all tests passed

**Comment by ypopovych** [comment]
@Ybrin @pixelmatrix guys, can we start the process of merging? I will need the updated library in my own libraries, and I don't want to fork it. It's better to update yours.

**Comment by pixelmatrix** [comment]
The Swift 5 stuff looks good to me. My only suggestion there is that Hashable is able to be computed for many custom structs now, so many of the custom hash(into:) methods may be unnecessary.

Personally I'd prefer this PR only introduces a single change (Swift 5 OR Keychain changes), but this is much cleaner than the previous one. I'm not sure how @Ybrin will feel about the Keychain stuff and splitting it into more submodules, but this seems fine to me.

**Comment by ypopovych** [comment]
About the logic behind Keychain extraction.

We have our own keychain implementation (with Rust based HD wallet). And in current realization, Web3 adds a dependency to `secp256k1.swift`, which we will never use. If we will split it, it can be used in both ways.

**Comment by ypopovych** [comment]
@Ybrin @pixelmatrix any news?

File: .travis.yml [hunk 1]
```diff
@@ -3,51 +3,42 @@
...
+      dist: trusty
+      # Generic language as travis doesn't know anything about SPM and Linux
+      language: generic
+      sudo: required
+      before_install:
+        - eval "$(curl -sL https://swiftenv.fuller.li/install.sh)"
+      script:
+        - swift build
+        - swift build -c release
+        - swift test
```
Comment (Ybrin) [inline .travis.yml]: Don't forget the trailing newlines please

File: Example/Tests/JsonTests/EthereumBlockObjectTests.swift [hunk 2]
```diff
@@ -185,48 +185,3 @@ class EthereumBlockObjectTests: QuickSpec {
         }
     }
 }
-
-extension EthereumBlockObject {
-
-    init(
```
Comment (Ybrin) [inline Example/Tests/JsonTests/EthereumBlockObjectTests.swift]: Are public initializers now automatically included in structs in Swift 5?

File: Package.swift [hunk 3]
```diff
@@ -11,29 +11,39 @@ let package = Package(
...
     ],
     targets: [
         .target(
             name: "Web3",
-            dependencies: ["BigInt", "CryptoSwift", "secp256k1"],
+            dependencies: ["BigInt", "CryptoSwift"],
             path: "Web3/Classes",
             sources: ["Core", "FoundationHTTP"]),
+        .target(
+            name: "Web3Keychain",
```
Comment (Ybrin) [inline Package.swift]: I forgot it. Why did you modularize this?

File: Web3/Classes/Core/Json/EthereumBlockObject.swift [hunk 4]
```diff
@@ -133,6 +133,35 @@ public struct EthereumBlockObject: Codable {
             case unsupportedType
         }
     }
+    
+    init(number: EthereumQuantity?, hash: EthereumData?, parentHash: EthereumData,
```
Comment (Ybrin) [inline Web3/Classes/Core/Json/EthereumBlockObject.swift]: I see. Forget the last comment about the struct initializers...

File: Web3/Classes/Core/Transaction/EthereumTransaction.swift [hunk 5]
```diff
@@ -202,37 +166,6 @@ public struct EthereumSignedTransaction {
             self.chainId = chainId
         }
     }
-    
-    // MARK: - Convenient functions
-
-    public func verifySignature() -> Bool {
```
Comment (Ybrin) [inline Web3/Classes/Core/Transaction/EthereumTransaction.swift]: Where is the verifySignature now?

**Comment by Ybrin** [comment]
@ypopovych

**Pull request closed without merging**